src/System.Slices/System/SpanExtensions_binary.cs

```diff
@@ -72,6 +102,21 @@ public static partial class SpanExtensionsLabs
             Span<T> castedSlice = slice.Cast<byte, T>();
             castedSlice[0] = value;
         }
+
+        /// <summary>
+        /// Writes a structure of type T into a slice of bytes.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool TryWrite<[Primitive]T>(this Span<byte> slice, T value)
+            where T : struct
+        {
+            if (Unsafe.SizeOf<T>() > (uint)slice.Length) {
+                return false;
+            }
+            Span<T> castedSlice = slice.Cast<byte, T>();
+            castedSlice[0] = value;
```

Comment:
Always write to index 0? Is it useful to have an overload which writes to index n?